src/ui/categories/list/CategoryTree.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Col, Spinner } from 'patternfly-react';
+import CategoryFolderIcon from 'ui/categories/common/CategoryFolderIcon';
+import CategoryExpandedIcon from 'ui/categories/common/CategoryExpandedIcon';
+import CategoryListMenuActions from 'ui/categories/list/CategoryListMenuActions';
+import RowSpinner from 'ui/pages/common/RowSpinner';
+
+class CategoryTree extends Component {
+  componentWillMount() {
+    if (this.props.onWillMount) this.props.onWillMount(this.props);
+  }
+
+  renderRows() {
+    const { categories } = this.props;
+    return categories.map((category, i) => {
+      const onClickExpand = () => {
+        if (!category.isEmpty) {
+          this.props.onExpandCategory(category.code);
+        }
+      };
+      const className = ['CategoryTree__tree-column-td'];
+      if (category.isEmpty) {
+        className.push('CategoryTree__tree-column-td--empty');
+      }
+
+      return (
+        <tr key={category.code} className="CategoryTree__row">
+          <td className={className.join(' ').trim()}>
```

Comment:
trim() is not necessary here (join(' ') only places spaces between array elements, it does not add leading or trailing spaces), please remove it